src/library/scala/collection/mutable/CollisionProofHashMap.scala

```diff
@@ -63,6 +64,9 @@ final class CollisionProofHashMap[K, V](initialCapacity: Int, loadFactor: Double
 
   @`inline` private[this] final def index(hash: Int) = hash & (table.length - 1)
 
+  override protected def fromSpecific(coll: IterableOnce[(K, V)] @uncheckedVariance): CollisionProofHashMap[K, V] @uncheckedVariance = CollisionProofHashMap.from(coll)
```

Comment:
Isn't the issue that the class extends `MapOps[K, V, Map, CollisionProofHashMap[K, V]]` instead of `MapOps[K, V, CollisionProofHashMap, CollisionProofHashMap[K, V]]` ?

Reply:
No, you need `MapCC = Map`. The methods from `MapOps` that build a `MapCC` for arbitrary `K2` and `V2` don't have the required `Ordering[K2]` to build a `CollisionProofHashMap` instead. You can only do that for `C = CollisionProofHashMap[K, V]` because you already have an `Ordering[K]`.

The problem here was that we inherit the `fromSpecific` and `newSpecificBuilder` implementations from `Map` and scalac doesn't realize that their return types are incorrect for `CollisionProofMap`.